packages/nodes-base/nodes/Kafka/Kafka.node.ts

```diff
@@ -220,6 +261,18 @@ export class Kafka implements INodeType {
 				message = this.getNodeParameter('message', i) as string;
 			}
 
+			if (useSchemaRegistry) {
+				try {
+					const schemaRegistryUrl = this.getNodeParameter('schemaRegistryUrl', 0) as string;
+					const eventName = this.getNodeParameter('eventName', 0) as string;
+
+					const registry = new SchemaRegistry({ host: schemaRegistryUrl });
+					const id = await registry.getLatestSchemaId(eventName);
+
+					message = await registry.encode(id, JSON.parse(message));
+				} catch (error) {}
```

Comment:
You can throw a `NodeOperationError` to keep the error display consistent. you can find an example in L287

Reply:
done